### PR TITLE
Draw line given slope 'b' and constant 'a'

### LINKED ISSUE #3: Draw a line based on best fit line given by regression calculator

### PR DESCRIPTION
**Pull request by shivaraj-bh**
+ Updated the options of Chart to have a fixed boundary of chart

### CODE REVIEW

File: js/Plotter.js [hunk 1]
```diff
@@ -1 +1,19 @@
-// Part C - Shivaraj
\ No newline at end of file
+// Part C - Shivaraj
```
Comment (LavishKS) [inline js/Plotter.js]:
```suggestion
```
Remove this comment

File: js/RegParams.js [hunk 2]
```diff
@@ -17,15 +17,15 @@ export function calculateRegressionParams () {
     const meanOfY = sumY/numberOfPointsOnCanvas;
 
     regressionPlot.data.datasets[0].data.forEach(ele => {
-        numerator += (ele.x - meanOfX)*(ele.y - meanOfY); // change it to a more meaningful variable name
-        denominator += (ele.x - meanOfX)*(ele.x - meanOfX); // change it to a more meaningful variable name
+        numerator += ele.x * (ele.y - meanOfY); // change it to a more meaningful variable name
```
Comment (LavishKS) [inline js/RegParams.js:19]: Can you paste the actual formula for this calculation here.

File: js/Tracker.js [hunk 3]
```diff
@@ -30,8 +32,11 @@ function addNewDatapoint(dataX, dataY, _nearbyElements) {
 
     // Add to the chart dataset.
     regressionData.push(dataPoint);
-
-    calculateRegressionParams ();
+    const numberOfPointsOnCanvas = regressionPlot.data.datasets[0].data.length;
+    if(numberOfPointsOnCanvas>1){
```
Comment (LavishKS) [inline js/Tracker.js]:
```suggestion
    if(regressionData.length>1){
```

File: js/Tracker.js [hunk 4]
```diff
@@ -30,8 +32,11 @@ function addNewDatapoint(dataX, dataY, _nearbyElements) {
 
     // Add to the chart dataset.
     regressionData.push(dataPoint);
-
-    calculateRegressionParams ();
+    const numberOfPointsOnCanvas = regressionPlot.data.datasets[0].data.length;
```
Comment (LavishKS) [inline js/Tracker.js]:
```suggestion
```

**Review by LavishKS — CHANGES_REQUESTED** [review]

File: js/RegParams.js (hunk 2 above)
Comment (sanya5549) [reply js/RegParams.js:19]: Using the earlier formula, I was able to match the regression params values with ones given in the link:
https://www.graphpad.com/quickcalcs/linear2/
Can @sbh69840 explain the reason for making the above change?

Comment (shivaraj-bh) [reply js/RegParams.js:19]: Okay, will paste the actual formula. Regarding @sanya5549 's comment, the formula was correct itself. I was facing some issue with plotter and I thought it was in calculating regression params, so I changed here. I resolved the plotter issue and in the end, thought this was lesser code, so kept it anyway.

Comment (LavishKS) [reply js/RegParams.js:19]: is this formula correct? or changes are required?

Comment (sanya5549) [reply js/RegParams.js:19]: will paste the formula when I make the changes.

**Review by sanya5549 — APPROVED** [review]

**Pull request merged**